### ISSUE TITLE
14.2.1 nested grid CH4 simulation doesn't write hourly SpeciesConcVV_CH4 properly

### ISSUE THREAD

**Issue by nicholasbalasus** [geoschem/geos-chem#1888]
### Name and Institution (Required)

Name: Nick Balasus
Institution: Harvard University

### Confirm you have reviewed the following documentation

- [x] [Support guidelines](https://geos-chem.readthedocs.io/en/stable/reference/SUPPORT.html)
- [x] [User manuals](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/related-docs.html)
- [x] [Debugging GEOS-Chem and HEMCO errors](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/debug-guide.html)
- [x] [Current and past Github issues](https://github.com/geoschem/geos-chem/issues)

### Description of your issue or question

For the IMI, we run nested grid methane simulations that write out concentrations hourly (instantaneous). We currently use 14.0.2. I have run a test simulation over the Europe nested grid domain for three days with recent versions of GEOS-Chem (14.1.0, 14.1.1, 14.2.0, and 14.2.1). All of them are working except for 14.2.1. The odd behavior is 14.2.1 only writes out SpeciesConcVV_CH4 for the first hour of each netCDF file (and the remaining 23 hours are just the fill value). This is presumably related to updates from HEMCO 3.7.0 to 3.7.1. This probably isn't a priority since 14.2.1 isn't an official release, but I'm happy to look for a fix if this sounds familiar to anyone or seems relevant to any GC 14.2.1/HEMCO 3.7.1 updates. There is nothing that looks wrong in the log file (attached) or the slurm output file (just times).

Code used to run simulation:
``` 
#!/bin/bash

# Environment file
source /n/home06/nbalasus/envs/gcclassic.rocky+gnu12.minimal.env

# Function to execute the commands for a specific version
function run_for_version {
    version="$1"
    cd ~
    mkdir "unmodified-gc-$version"
    cd "unmodified-gc-$version"
    git clone https://github.com/geoschem/GCClassic.git
    cd GCClassic
    if [[ $version == *"14.2"* ]]; then
        git checkout "dev/$version"
    else
        git checkout "$version"
    fi
    git submodule update --init --recursive
    cd run
    # Choose CH4, nested grid over EU, MERRA2 0.5 x 0.625, 47 levels
    c="3\n1\n3\n3\n2\n~/unmodified-gc-$version\n\nn\n"
    printf ${c} | ./createRunDir.sh
    cd ~/unmodified-gc-$version/gc_05x0625_EU_47L_merra2_CH4/build
    cmake ../CodeDir -DRUNDIR=..
    make -j
    make install
    cd ..
    # Write out SpeciesConcCH4_VV every hour (one netCDF file per day) for [20190101 to 20190103]
    sed -i -e "s|SpeciesConc.frequency:      00000100 000000|SpeciesConc.frequency:      00000000 010000|g" \
        -e "s|SpeciesConc.duration:       00000100 000000|SpeciesConc.duration:       00000001 000000|g" \
        -e "s|SpeciesConc.mode:           'time-averaged'|SpeciesConc.mode:           'instantaneous'|g" \
        -e "s|'CH4'|#'CH4'|g" \
        -e "s|'Metrics'|#'Metrics'|g" \
        -e "s|'SpeciesConcMND_?ALL?|#'SpeciesConcMND_?ALL?|g" HISTORY.rc
    sed -i -e 's|$ROOT/SAMPLE_BCs/v2021-07/CH4|/n/holylfs05/LABS/jacob_lab/imi/ch4/tropomi-boundary-conditions/v2023-06|g' HEMCO_Config.rc
    sed -i -e "s|20190201|20190103|g" geoschem_config.yml
    # Copy run script and use it
    if [[ $version == "14.0.2" || $version == "14.1.0" ]]; then
        cp runScriptSamples/geoschem.run .
    else
        cp runScriptSamples/operational_examples/harvard_cannon/geoschem.run .
    fi
    sbatch geoschem.run
}

# Loop over the versions
for version in "14.0.2" "14.1.0" "14.1.1" "14.2.0" "14.2.1"; do
    run_for_version "$version"
done
```

Code to plot:
```
import matplotlib.pyplot as plt
import cartopy.crs as ccrs
import xarray as xr

fig,axs = plt.subplots(5,2,figsize=(13,15),subplot_kw={"projection":ccrs.PlateCarree()})

for idx,version in enumerate(["14.0.2","14.1.0","14.1.1","14.2.0","14.2.1"]):#,"14.1.1","14.2.0","14.2.1"]):
    with xr.open_dataset(f"~/unmodified-gc-{version}/gc_05x0625_EU_47L_merra2_CH4/OutputDir/GEOSChem.SpeciesConc.20190102_0000z.nc4") as ds:

        if version == "14.0.2":
            var = "SpeciesConc_CH4"
        else:
            var = "SpeciesConcVV_CH4"

        (ds[var].isel(time=0,lev=0)*1e9).plot(ax=axs[idx,0],vmin=1800,vmax=2400)
        axs[idx,0].set_title(f"{version}, t=0")
        (ds[var].isel(time=1,lev=0)*1e9).plot(ax=axs[idx,1],vmin=1800,vmax=2400)
        axs[idx,1].set_title(f"{version}, t=1")

for ax in axs.flatten():
    ax.coastlines()
```

Plot:
![image](https://github.com/geoschem/geos-chem/assets/91984611/078c10f1-e09c-4a5a-884c-d968e8d0302b)

[GC-14.2.1.log](https://github.com/geoschem/geos-chem/files/12172709/GC-14.2.1.log)

**Comment by yantosca** [comment]
Thanks @nicholasbalasus.  BTW, in the 14.2.1 integration tests I made sure that each update to 14.2.1 was identical to the previous update. 

We recently updated the netCDF interface to Fortran-90 (as required for merging GEOS-Chem and HEMCO into CESM.  I doubt this could have caused the issue but you never know.

Which compiler did you use: ifort or gfortran?

I'll see if I can reproduce your results.

Tagging @msulprizio @lizziel

**Comment by nicholasbalasus** [comment]
Thanks, Bob! I used `gcclassic.rocky+gnu12.minimal.env` (consistent with [this](https://github.com/Harvard-ACMG/cannon-env/blob/main/envs/gnu12/gcclassic.rocky%2Bgnu12.minimal.env)).

**Comment by yantosca** [comment]
@nicholasbalasus: Can you also try with `gcclassic.rocky+gnu10.minimal.env`?  This is the compiler I used to run the integration tests.  Just to rule out that it is something specific to gnu12 and not gnu10.

**Comment by nicholasbalasus** [comment]
@yantosca I've just finished running with `gcclassic.rocky+gnu10.minimal.env` and the result is identical. Could the behavior be related to `instantaneous` since the integration tests use `time-averaged`?

**Comment by yantosca** [comment]
@nicholasbalasus: It could be.  Something to dig into here.

**Comment by nicholasbalasus** [comment]
Thank you, @yantosca. I just tested this with Haipeng's environment file (`/n/home04/hplin/rcfiles/rocky_gc_hplin_23.env`) and the problem is still present.

**Comment by yantosca** [comment]
@nicholasbalasus: This might be related to the netcdf-F90 updates.  Still testing here.

**Comment by yantosca** [comment]
@nicholasbalasus @jimmielin: I've found a fix!  This is really obscure behavior, but with the netCDF-F90 interface, if you have spatial data for a single time slice, then you have to pass that to `NF90_Put_Var` as a 4D array with a singleton 4th dimension instead of as a 3d array (which is what we were doing).  The netCDF-F77 library evidently didn't care and would let you pass a 3-D array.  Fix forthcoming...

**Comment by nicholasbalasus** [comment]
This is awesome, thanks @yantosca! I can test this as above.

**Comment by nicholasbalasus** [comment]
Looks great!

![image](https://github.com/geoschem/geos-chem/assets/91984611/c692bdde-ec41-449f-a274-3f0a0ed4b473)
[run.txt](https://github.com/geoschem/geos-chem/files/12228397/run.txt)

**Comment by yantosca** [comment]
Thanks @nicholasbalasus.  I'll make a PR for this ASAP.

**Comment by yantosca** [comment]
We can now close this issue, as PR #1896 is now merged into the 14.2.1 development stream.  Thanks again @nicholasbalasus!
